File: java/src/main/java/com/myriad/wdl/model/expressions/WdlBooleanLiteral.java
package com.myriad.wdl.model.expressions;

public final class WdlBooleanLiteral extends WdlLiteral<Boolean> {
  public WdlBooleanLiteral() {}

  public WdlBooleanLiteral(boolean value) {
    super(value);
  }

  @Override
  public ComponentType componentType() {
    return ComponentType.BOOL_LIT;
  }
}

File: typescript/src/model/expressions/wdl-boolean-literal.ts
/** Boolean literal expression nodes. */
import { WdlExpressionComponentType } from './wdl-expression.js';
import { WdlLiteral } from './wdl-literal.js';

/** Models a `true` or `false` literal. */
export class WdlBooleanLiteral extends WdlLiteral<boolean> {
  /** Returns the broad expression family for this node. */
  public componentType(): WdlExpressionComponentType {
    return WdlExpressionComponentType.BOOL_LIT;
  }
}
